src/test/java/org/junit/internal/runners/statements/ExpectExceptionTest.java

```diff
@@ -0,0 +1,29 @@
+package org.junit.internal.runners.statements;
+
+import org.junit.Test;
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.runners.model.Statement;
+
+import static org.junit.Assert.*;
+
+public class ExpectExceptionTest {
+
+    @Test
+    public void givenWeAreExpectingAssumptionViolatedExceptionAndAStatementThrowingAssumptionViolatedException() {
```

Comment:
I like descriptive method names but this is a bit long. How about:

`whenExpectingAssumptionViolatedExceptionStatementsThrowingItShouldPass`